string[] action_types
string[] topics
float32[] duration_ms
---
string error